how-to/how-to-use-k8s-with-cri-containerd-and-kata.md

```diff
@@ -0,0 +1,160 @@
+# How to use Kata Containers and CRI (containerd plugin)
+
+This document describes how to set up a single-machine Kubernetes cluster. 
+The Kuberentes cluster will use the CRI containerd plugin and Kata Containers to lanuch untrsuted workloads.
+
+## Requirements 
+- Kuberentes, kubelet, kubeadm
+- cri-containerd
+- Kata Containers
+
+For information about the supported version of this components see 
+Kata Containers [versions.yaml](https://github.com/kata-containers/runtime/blob/master/versions.yaml) file.
+
+## Install containerd(with CRI plugin enabled)
+
+Follow the instructions from [CRI installation guide](http://github.com/containerd/cri/blob/master/docs/installation.md)
+
+## Install Kata Containers / Kata
+
+Follow the instructions to [install Kata](https://github.com/kata-containers/documentation/blob/master/install/README.md).
+
+## Install Kubernetes
+Install Kuberentes in your host.  See kubeadm [installation](https://kubernetes.io/docs/tasks/tools/install-kubeadm/)
+
+### Configure contaienrd to use Kata Containers
+
+The CRI contaienrd plugin support a define two runtimes.
+
+- default runtime: A runtime that is used by default to run workloads.
+- Untrusted workload runtime: A runtime that will be used run untrusted workloads.
+
+#### Define the Kata runtime as `untrusted_workload_runtime` 
+
+Configure the Kata runtime for untrusted workload with the [config option](https://github.com/containerd/cri/blob/v1.0.0-rc.0/docs/config.md)
+`plugins.cri.containerd.untrusted_workload_runtime`.
+
+Unless configured otherwise, the default runtime is set to runc.
+```bash
+$ sudo mkdir -p /etc/containerd/
+$ cat << EOT | sudo tee /etc/containerd/config.toml
+[plugins]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.untrusted_workload_runtime]
+        runtime_type = "io.containerd.runtime.v1.linux"
+        runtime_engine = "/usr/bin/kata-runtime"
+EOT
+```
+
+### Configure Kubelet to use containerd
+
+In order to allow kubelet use containerd (using CRI interface)  configure the service to
+point to containerd socket.
+
+```bash
+# Configure k8s to use containerd
+$ sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
+$ cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+[Service]                                                 
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+EOF
+$ sudo systemctl daemon-reload
+```
+
+### Optional: Configure proxy
+
+If you are behind a proxy this script will configure your proxy for docker
+kubelet and containerd.
+
+```bash
+# Set proxys
+$services=(
```

Comment:
Nice! https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh is causing Travis to fail this doc due to the missing space between `$` and `services=(` here:

```
$ services=(
```

Reply:
If you re-add that space and respin Travis, we should be good to get this landed.